gemrb/core/DialogHandler.cpp

```diff
@@ -100,16 +100,15 @@ bool DialogHandler::InitDialog(Scriptable* spk, Scriptable* tgt, const ResRef& d
 	//target is here because it could be changed when a dialog runs onto
 	//and external link, we need to find the new target (whose dialog was
 	//linked to)
-
 	Actor *oldTarget = GetLocalActorByGlobalID(targetID);
-	speakerID = spk->GetGlobalID();
-	targetID = tgt->GetGlobalID();
+	ScriptID spkID = SetSpeaker(spk);
+	targetID = SetTarget(tgt);
```

Comment:
These are not modified later (we don't need a backup), so it could just be:
```suggestion
	SetSpeaker(spk);
	SetTarget(tgt);
```
(with the other spkID change reverted)